Configure

```diff
@@ -3280,33 +3293,74 @@ sub usage
         {
         print STDERR $usage;
         print STDERR "\npick os/compiler from:\n";
-        my $j=0;
         my $i;
         my $k=0;
+        print STDERR "  ";
         foreach $i (sort keys %table)
                 {
                 next if $table{$i}->{template};
                 next if $i =~ /^debug/;
-                $k += length($i) + 1;
+                $k += length($i);
                 if ($k > 78)
                         {
-                        print STDERR "\n";
-                        $k=length($i);
+                        print STDERR "\n  ";
+                        $k=length($i) + 2;
                         }
+                $k += 1;
                 print STDERR $i . " ";
                 }
         foreach $i (sort keys %table)
                 {
                 next if $table{$i}->{template};
                 next if $i !~ /^debug/;
-                $k += length($i) + 1;
+                $k += length($i);
                 if ($k > 78)
                         {
-                        print STDERR "\n";
-                        $k=length($i);
+                        print STDERR "\n  ";
+                        $k=length($i) + 2;
                         }
+                $k += 1;
                 print STDERR $i . " ";
                 }
+        print STDERR "\n";
+
+        disable();
+
+        # Do some color wizardry.  Ideally, this would be done by a library and
+        # would use proper termcap/terminfo, but we would rather avoid external
+        # dependencies.  It's disabled by default unless you define CLICOLOR in
+        # your env.  (We approximate ls(1)'s behavior in BSD in this regard.)
+        my $en_color="";
+        my $dis_color="";
+        my $reset_color="";
+        my $this_color;
+        if ( defined $ENV{'CLICOLOR'} and (defined $ENV{'CLICOLOR_FORCE'} or -t 2) )
```

Comment:
I don't think this is quite right.

Surely CLICOLOR_FORCE=1 should take precedence even if CLICOLOR is not set. Also CLICOLOR_FORCE=0 will not work correctly.

Do you have a link to a reference for the CLICOLOR variable?

It might be better to make the colorization support a separate PR. The uppercase/lowercase thing is fine here though.

There is also the NO_COLOR variable "standard" which is rising in popularity and which I was aware of unlike this (competing?) proposal. Hmm...

Reply:
> Surely CLICOLOR_FORCE=1 should take precedence even if CLICOLOR is not set. Also CLICOLOR_FORCE=0 will not work correctly.

You're right, it seemed a bit wonky to me, too, but that's what the random man page I found said to do.  Furthermore, it was unambiguous that `CLICOLOR_FORCE=0` has the same effect as `CLICOLOR_FORCE=1`.  :shrug:

Ultimately, there is a real [XKCD "Standards"](https://xkcd.com/927/) situation going when it comes to CLI color support, and the original benefit of `CLICOLOR` was that it let me default colors to _off_, which felt like the safest play given the wide range of systems on which openssl might run or compile.  If you think enabling color by default on a terminal is OK, then how about using [these rules](http://bixense.com/clicolors/), to wit:

```suggestion
        if ( !defined $ENV{'NO_COLOR} and (defined $ENV{'CLICOLOR_FORCE'} or -t 2) )
```